src/FixerDescriptionAwareInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ */
+interface FixerDescriptionAwareInterface
+{
+    /**
+     * @return FixerDescription
+     */
+    public function getFixerDescription();
```

Comment:
So, you would, comparing to our master branch:
- modify existing `FixerInterface`
- add new `FixerDescriptionAwareInterface`

What's the benefits here?

---------------------

2 ways to go:
Fist one: keep existing interface not modified. Add new one for "described" fixers.
Second one: modify existing interface to have that `getDescriptionDTO` method obligatory.

Introducing new interface while manipulating existing one will still be a big BC breaker.
`FixerDescriptionAwareInterface` was introduced to not break BC when we already have a RC version and `2@stable` is close to release.

I totally agree that putting a `getDesc...DTO...` in main interface and enforcing it everywhere would be really nice option, I'm just afraid that it's too late for doing that for 2.x line. That would mean that all custom fixers need to be rewritten for 2.x version.